contrib/dimfeld/httptreemux.v5/httptreemux.go

```diff
@@ -115,17 +115,40 @@ func getRoute(router *httptreemux.TreeMux, w http.ResponseWriter, req *http.Requ
 
 	// Check for redirecting route due to trailing slash for parameters.
 	// The redirecting behaviour originates from httptreemux router.
-	if lr.StatusCode == http.StatusMovedPermanently && strings.HasSuffix(route, "/") {
-		rReq := req.Clone(req.Context())
-		rReq.RequestURI = strings.TrimSuffix(rReq.RequestURI, "/")
-		rReq.URL.Path = strings.TrimSuffix(rReq.URL.Path, "/")
-
-		lr, found = router.Lookup(w, rReq)
-		if !found {
-			return "", false
+	// TODO: what about the other two router.RedirectBehavior cases?
+	if lr.StatusCode == http.StatusMovedPermanently && router.RedirectBehavior == httptreemux.Redirect301 {
+		if strings.HasSuffix(route, "/") {
+			// if the route has a trailing slash, retry lookup without trailing slash
+			rReq := req.Clone(req.Context())
+			rReq.RequestURI = strings.TrimSuffix(rReq.RequestURI, "/")
+			rReq.URL.Path = strings.TrimSuffix(rReq.URL.Path, "/")
+
+			lr, found = router.Lookup(w, rReq)
+			if !found {
+				return "", false
+			}
+		} else {
+			// if the route does not have a trailing slash, retry lookup without trailing slash
+			rReq := req.Clone(req.Context())
+			rReq.RequestURI = rReq.RequestURI + "/"
+			rReq.URL.Path = rReq.URL.Path + "/"
+
+			lr, found = router.Lookup(w, rReq)
+			if !found {
+				return "", false
+			}
```

Comment:
I'm a little confused by this. The comment says if the route does not have a trailing slash, to retry the lookup without the trailing slash, but this code looks like it's adding the slash

Reply:
Eep! Yep. That's definitely a typo. Thanks for pointing it out!